src/main/java/org/alliancegenome/curation_api/dao/base/BaseSQLDAO.java

```diff
@@ -111,6 +111,22 @@ public E find(Long id) {
 			return null;
 		}
 	}
+	
+	public List<String> findFilteredIds(Map<String, Object> params) {
+		List<String> primaryKeys = new ArrayList<>();
+		SearchResponse<E> results = findByParams(params);
+		for (E entity : results.getResults()) {
+			String pkString;
+			try {
+				pkString = (String) entityManager.getEntityManagerFactory().getPersistenceUnitUtil().getIdentifier(entity);
+			} catch (ClassCastException e) {
+				pkString = Long.toString((Long) entityManager.getEntityManagerFactory().getPersistenceUnitUtil().getIdentifier(entity));
```

Comment:
May want to catch any errors here also.

Reply:
What would we do if we caught the exception?  As you say, I can't think of any other types of PK we're going to encounter.

Reply:
So I guess maybe a 
`String pkString = null;`

and later:

if(phString != null) {
primaryKeys.add(pkString);
}

That way if it does fail... things can still move forward maybe this would happen anyway not sure.

Reply:
Done